daemon/boardd/server.cpp

```diff
@@ -165,14 +165,15 @@ void start_server(const char *host, unsigned short port) {
     std::cerr << "boardd: built with mulit-threading, "
 	<< NUM_THREADS << " threads." << std::endl;
 
-    std::list<std::thread> threads;
     asio::io_service io_service;
     Server server(io_service, host, port);
-    for (int i = 0; i < NUM_THREADS; ++i) {
-	threads.push_back(std::move(
-		std::thread(std::bind(&ServiceThread, std::ref(io_service)))));
+    std::vector<std::future> futures;
+    futures.reserve(NUM_THREADS);
+    for (std::size_t i = 0; i < NUM_THREADS; ++i) {
+	futures.emplace_back(
+		std::async(std::launch::async,ServiceThread,std::ref(io_service)));
```

Comment:
I digged into documentation about [std::async](http://en.cppreference.com/w/cpp/thread/async). It doesn't guaranteed to create the same number of kernel threads.
> The template function async runs the function f asynchronously (potentially in a separate thread which may be part of a thread pool) and returns a std::future that will eventually hold the result of that function call.

We usually set the number of threads greater than the number of cores we have, due to the application is usually IO-bounded. We need to have control on how many kernel threads the application creates.